src/init.cpp

```diff
@@ -122,6 +122,13 @@ struct parseFromString<SizeType> {
   };
 };
 
+template <>
+struct parseFromString<bool> {
+  static std::size_t call(const std::string& var) {
+    return !var.empty();
```

Comment:
`env DEBUG_DUMP_TRISOLVER_DATA=OFF` will set `debug_dump_trisolver_data` to `true`, correct?

Reply:
Yeah, I thought a bit about that. Would you prefer something like this?

```cpp
template <>
struct parseFromString<bool> {
  static std::size_t call(std::string var) {
      constexpr const char* true_values[] = {"ON", "TRUE", "YES"};
      constexpr std::size_t n = sizeof(true_values)/sizeof(const char*);
      std::transform(var.cbegin(), var.cend(), var.begin(), [](unsigned char c){ return std::toupper(c); });
      bool is_true = (true_values + n) != std::find(true_values, true_values + n, str);
      return is_true;
  }
};
```

Reply:
I think so, yes. I would maybe add a `1` in there as well. For that version you could maybe instead this for the array
```
constexpr const std::array true_values{"ON", "TRUE", "YES", "1"};
```

However, the alternative to the above would be to treat `DEBUG_DUMP_TRISOLVER_DATA` being set as true, which is maybe what you were going for? If yes, that'd also be fine by me, but that requires some other changes outside of `parseFromString` (or just always returning `true` for `bool` with the assumption that `parseFromString` is only called if there is an environment variable defined; I don't like this assumption though...).

Reply:
We had a quick chat about this and I just want to document some of the options that we discussed:

| option             | a     | b     | c     | d     |
|--------------------|-------|-------|-------|-------|
| unset DEBUG        | false | false | false | false |
| export DEBUG=      | true  | false | false | true  |
| export DEBUG=TRUE  | true  | true  | true  | true  |
| export DEBUG=FALSE | true  | false | true  | false |

The current implementation on this branch is `c`. The suggestion by @albestro is `b`. `a` is "if the environment variable is set the option is true", and `d` is a mix: "if the environment variable is set to empty, or a true-ish non-empty value the option is true".

Reply:
In order to raise a point, I would suggest a small change to the first row of the above table:

| option             | a     | b     | c     | d     |
|--------------------|-------|-------|-------|-------|
| unset DEBUG        | default | default | default | default |
| export DEBUG       | true  | false | false | true  |
| export DEBUG=TRUE  | true  | true  | true  | true  |
| export DEBUG=FALSE | true  | false | true  | false |

One point of giving meaning to the value of the variable is that user will have the chance to set the option as preferred and not just `{default, true}`, which in the case `default=true` it means that the variable is meaningless.

For this reason I would drop A. Moreover, I would drop also C because it looks counter-intuitive at first sight (DEBUG=FALSE -> TRUE).

Finally, I would go for B or D. Actually D looks the more reasonable one to me too (probably it was also what @msimberg was supporting).

@rasolca do you agree? do you have any preference?

Reply:
Either b or d.

But I would suggest `e`:
`true_values{"ON", "TRUE", "YES", "1"};` `false_values{"OFF", "FALSE", "NO", "0"};` and the rest default.

Reply:
@rasolca proposes E

| option |  b | d | e |
| -- |  -- | -- | --  |
| unset DEBUG |  default  | default | default |
| export DEBUG |  false | true | default |
| export DEBUG=TRUE |  true | true | true |
| export DEBUG=FALSE  | false | false | false |
| export DEBUG=*  | ? | ? | default | 

that can be described as "the variable should be interpreted as true-ish or false-ish, otherwise default"

Reply:
Just opened #913. Let's set the discussion there, merge it and then we can use it here 😉